### PR TITLE
fix: Support scoped package names for initialising projects

### LINKED ISSUE #4416: Does not support templates with scoped package name
### Summary

I have created templates for `realm`, `@realm/expo-template-js` and `@realm/expo-template-ts`.  Unfortunately, when I try to initialize a project with these templates, the cli tool crashes:
```
expo init ExpoJsApp -t @realm/expo-template-js
Error: ENOENT: no such file or directory, open '/Users/user/.expo/template-cache/@realm/expo-template-js-0.3.0.tgz'
⠋ Downloading template.%
```

The cli stores the template cache in: 
```
 '/Users/user/.expo/template-cache/realm-expo-template-js-0.3.0.tgz'
```

It appears that the cli tool is not resolving scoped namespace packages correctly.

### Environment

  expo-env-info 1.0.3 environment info:
    System:
      OS: macOS 12.4
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.14.0 - ~/.nvm/versions/node/v16.14.0/bin/node
      Yarn: 1.22.17 - ~/.nvm/versions/node/v16.14.0/bin/yarn
      npm: 7.24.2 - ~/.nvm/versions/node/v16.14.0/bin/npm
      Watchman: 2022.03.14.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.11.2 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
      Android SDK:
        API Levels: 28, 29, 30, 31
        Build Tools: 29.0.2, 30.0.2, 30.0.3, 31.0.0
        System Images: android-28 | Google APIs Intel x86 Atom, android-29 | Google APIs Intel x86 Atom, android-30 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom
        Android NDK: 21.4.7075529
    IDEs:
      Android Studio: 2020.3 AI-203.7717.56.2031.7678000
      Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
    npmGlobalPackages:
      expo-cli: 5.4.9
    Expo Workflow: managed

### Please specify your device/emulator/simulator platform, model and version

all

### Error output

```
expo init ExpoJsApp -t @realm/expo-template-js
Error: ENOENT: no such file or directory, open '/Users/user/.expo/template-cache/@realm/expo-template-js-0.3.0.tgz'
⠋ Downloading template.%
```

### Reproducible demo or steps to reproduce from a blank project

Just initialize a project with our template:
```
expo init ExpoJsApp -t @realm/expo-template-js
```

### PR DESCRIPTION
**Pull request by takameyer**
# Why
The `npm pack` produces a tgz file and shows the path with the `filename` property of the command output.
Unfortunately for scoped package names, this is outputting an incorrect filename.  This is causing initialising
expo projects from a template with a scoped package name to fail.

This is due to a bug in npm (https://github.com/npm/cli/issues/3405)

This PR applies a workaround.

This closes #4416

# How
The fix checks if the package name and the filename from the `npm pack` command start with `@`.  If so, then it removes the `@` and replaces `/` with `-` from the `npm pack` filename output.   It's checking both package name and filename, as it is possible that npm will fix their bug at some time in the future.

# Test Plan
To manually test this, use the following command:
```
expo init RealmJsApp -t @realm/expo-template-js
```

### CODE REVIEW

**Review by wkozyra95 — APPROVED** [review]
LGTM, but I would prefer to wait for Evan's review before merging that

**Review by EvanBacon — APPROVED** [review]
LGTM

**Pull request merged**
